helpdesk/models.py

```diff
@@ -577,19 +577,13 @@ def _get_status(self):
     def _absolute_uri(self, relative):
         """
         Returns an absolute URL for the given relative URL.  This will
-        use self.request, if available (using the same protocol and
-        domain as from the original request), otherwise will make a
-        best guess from the Site or else just a hard-coded domain.
+        make a best guess from the Site or else just a hard-coded domain.
         """
-        # If we have a request object, use it to construct the absolute URL
-        if self.request:
-            return self.request.build_absolute_uri(relative)
-        else:
-            try:
-                site = Site.objects.get_current()
-            except:
-                site = Site(domain='configure-django-sites.com')
-            return u"http://%s%s" % (site.domain, relative)
+        try:
+            site = Site.objects.get_current()
+        except:
+            site = Site(domain='configure-django-sites.com')
```

Comment:
"configure-django-sites.com", really? Is that just an attempt to embed a little bit of documentation into the eventual error message? Like: "Site not found: configure-django-sites.com"... leads to "Oh, I guess I'd better configure django sites!" ??

Admittedly, I've been known to do my fair share of this kind of thing, (in code where I can't control the error message directly, but I'd like to put some information in it).

But at the very least, I would have put in a good comment explaining what was going on.

Anyway, I know this wasn't our code originally. But I also wonder if we wouldn't want to have an actual failure here, (so that _we_ get an actual error instead of the user silently getting a bogus domain).

Reply:
I'll change this to raise an exception instead. Thanks for the feedback!